api/public_link_manager_owncloud/public_link_manager_owncloud.go

```diff
@@ -195,24 +196,34 @@ func (lm *linkManager) CreatePublicLink(ctx context.Context, path string, opt *a
 		}
 	}
 
-	fileSource, err := strconv.ParseUint(itemSource, 10, 64)
+	created := time.Unix(int64(time.Now().Unix()), 0)
+
+	tx, err := lm.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+	result, err := tx.Exec("INSERT INTO share_ids () VALUES ()")
+	if err != nil {
+		l.Error("", zap.Error(err))
+		return nil, err
+	}
+	lastId, err := result.LastInsertId()
 	if err != nil {
 		l.Error("", zap.Error(err))
 		return nil, err
 	}
 
-	shareName := gopath.Base(path)
-
-	stmtString := "insert into oc_share set share_type=?,uid_owner=?,uid_initiator=?,item_type=?,fileid_prefix=?,item_source=?,file_source=?,permissions=?,stime=?,token=?,share_name=?"
-	stmtValues := []interface{}{3, u.AccountId, u.AccountId, itemType, prefix, itemSource, fileSource, permissions, time.Now().Unix(), token, shareName}
+	// This is incorrect for projects... The owner should be the service account
+	stmtString := "INSERT INTO public_links SET id=?,created_at=?,updated_at=?,uid_owner=?,uid_initiator=?,item_type=?,initial_path=?,inode=?,instance=?,permissions=?,orphan=?,token=?,quicklink=?,notify_uploads=?"
```

Comment:
`notify_uploads` is handled but not `notify_uploads_extra_recipients`

Reply:
The value is set to 0, which would be the default (we cannto send notifications from old system). So it's fine (or I would just remove that and keep the default).